facilitator/src/config.rs

```diff
@@ -44,7 +44,7 @@ impl S3Path {
 }
 
 impl Display for S3Path {
-    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> fmt::Result {
```

Comment:
Is this change fine? I had to do it since `export:Formatter` is no longer public in the `serde` crate.

Reply:
This is interesting! OK, so [`std::fmt::Display::fmt`](https://doc.rust-lang.org/std/fmt/trait.Display.html#tymethod.fmt) wants a [`std::fmt::Formatter`](https://doc.rust-lang.org/std/fmt/struct.Formatter.html). That's a concrete struct, not a trait, so how did this ever work? Intuitively the compiler should have rejected the `serde::export::Formatter`. I think what happened is that Serde used to have a module `export`, which [they removed recently](https://github.com/serde-rs/serde/commit/dd1f4b483ee204d58465064f6e5bf5a457543b54#diff-110ad2e16d30874ce5f0193e158b33b11289693a8ac60024f774317b92f4f857). That module contained the line `pub use lib::fmt::{self, Formatter};`, which re-exported `std::fmt::Formatter` as `serde::export::Formatter`. Finally it's `lib::fmt::Formatter` because [Serde's `lib.rs` re-exports all of `std::*` as `lib::*` within the Serde crate](https://github.com/serde-rs/serde/blob/b0c99ed761d638f2ca2f0437522e6c35ad254d93/serde/src/lib.rs#L152).

tl;dr: it was always wrong to use `serde::export::Formatter` and we were just lucky that it happened to alias to `std::fmt::Formatter`, so your change is correct. I would just add `Formatter` to the existing `use std::fmt{}` block at the top of this file so you don't need the fully qualified type name in the signatures of all the `fmt()` methods.

Reply:
Awesome! Thank you for this!